homeassistant/components/evohome/__init__.py

```diff
@@ -352,8 +502,26 @@ def __init__(self, evo_broker, evo_device) -> None:
         self._device_state_attrs = {}
 
     @callback
-    def _refresh(self) -> None:
-        self.async_schedule_update_ha_state(force_refresh=True)
+    def _refresh(self, payload=None) -> None:
+        if payload is None:
+            self.async_schedule_update_ha_state(force_refresh=True)
+        elif payload["unique_id"] == self._unique_id:
```

Comment:
Can you flatten these if's a bit by throwing in some `return` when you know that you're done.